src/main/java/com/simanavets/tenderflexbackend/service/UserService.java

```diff
@@ -0,0 +1,40 @@
+package com.simanavets.tenderflexbackend.service;
+
+import com.simanavets.tenderflexbackend.entity.User;
+import com.simanavets.tenderflexbackend.repository.UserRepository;
+//import org.springframework.security.core.userdetails.UserDetails;
+//import org.springframework.security.core.userdetails.UserDetailsService;
+//import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+
+@Service
+public class UserService /*implements UserDetailsService*/ {
+
+    private final UserRepository userRepository;
+
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public List<User> findAllUsers() {
+        return userRepository.findAll();
+    }
+
+    public void createUser(User user) {
+        userRepository.save(user);
```

Comment:
We cannot just save entity which we got from the Client side. It must pass two steps:

1. Validation: check if all field you need are exist and has valid values
2. Transformation: transform User to DAO (Data Access Object). It is good practice for future, it because sometime some data not needed, some data must be transformed and some data must be attached from another source.

Reply:
Is it correct to transform User to DAO , not to DTO?